agency/detail/optional.hpp

```diff
@@ -70,7 +70,9 @@ inline void throw_bad_optional_access(const char* what_arg)
 }
 
 
+#if !(defined(__clang__) && defined(__CUDA__))
```

Comment:
I think the idea is to hide `#pragma nv_exec_check_disable` from compilers which aren't nvcc. This is the same thing these lines intend to do in `tuple_impl.hpp`.

I'm not exactly sure what the correct incantation is, but I feel like it should be equivalent to the thing we were doing in `on_chip_cast`. We basically want to hide nvcc-specific code from other compilers.